packages/experiments/src/components/ExtendedPicker/BaseFloatingPicker.tsx

```diff
@@ -0,0 +1,478 @@
+  protected currentPromise: PromiseLike<any>;
+
+  constructor(basePickerProps: P) {
+    super(basePickerProps);
+
+    this.suggestionStore = new SuggestionsController<T>();
+    this.state = {
+      queryString: '',
+      suggestedDisplayValue: '',
+      isMostRecentlyUsedVisible: false,
+      moreSuggestionsAvailable: false,
+      isSearching: false,
+      didBind: false,
+    };
+  }
+
+  public get isSuggestionsShown(): boolean {
+    return this.state.suggestionsVisible ? false : this.state.suggestionsVisible as boolean;
+  }
+
+  public get selectedSuggestionAlert(): string | undefined {
+    const currentIndex = this.suggestionStore.currentIndex;
+    const selectedSuggestion = currentIndex > -1 ? this.suggestionStore.getSuggestionAtIndex(this.suggestionStore.currentIndex) : undefined;
+    const selectedSuggestionAlert = selectedSuggestion ? selectedSuggestion.ariaLabel : undefined;
+
+    return selectedSuggestionAlert;
+  }
+
+  public onQueryStringChanged(queryString: string) {
+    if (queryString !== this.state.queryString) {
+      this.setState({
+        queryString: queryString,
+        moreSuggestionsAvailable: true,
+        isMostRecentlyUsedVisible: false,
+        suggestionsVisible: true,
+      });
+      if (queryString == '') {
+        this.updateSuggestionWithZeroState();
+      } else {
+        this.updateValue(queryString);
+      }
+
+    }
+  }
+
+  public hidePicker() {
+    this.setState({
+      suggestionsVisible: false,
+    });
+  }
+
+  public showPicker() {
+    this.setState({
+      suggestionsVisible: true,
+    });
+
+    if (this.state.queryString == '') {
+      this.updateSuggestionWithZeroState();
+    } else {
+      this.updateValue(this.state.queryString);
+    }
+  }
+
+  public componentDidMount() {
+    this._bindToInputElement();
+  }
+
+  public componentDidUpdate() {
+    this._bindToInputElement();
+  }
+
+  public componentWillUnmount() {
+    this._unbindFromInputElement();
+  }
+
+  @autobind
+  public dismissSuggestions() {
+    this.setState({ suggestionsVisible: false });
+  }
+
+  public completeSuggestion() {
+    if (this.suggestionStore.hasSelectedSuggestion()) {
+      this.onChange(this.suggestionStore.currentSuggestion!.item);
+    }
+  }
+
+  @autobind
+  public refocusSuggestions(keyCode: KeyCodes) {
+    if (keyCode === KeyCodes.up) {
+      this.suggestionStore.setSelectedSuggestion(
+        this.suggestionStore.suggestions.length - 1
+      );
+    } else if (keyCode === KeyCodes.down) {
+      this.suggestionStore.setSelectedSuggestion(0);
+    }
+  }
+
+  public render() {
+    let { className } = this.props;
+    return (
+      <div
+        ref={ this._resolveRef('root') }
+        className={ css('ms-BasePicker', className ? className : '') }
+      >
+        { this.renderSuggestions() }
+      </div>
+    );
+  }
+  protected renderSuggestions(): JSX.Element | null {
+    let TypedSuggestion = this.SuggestionOfProperType;
```

Comment:
Instead of copying part of BasePicker, why not just separate it into a new component from BasePicker? It shouldn't impact its current behavior by doing that.

Reply:
I created ExtendedPicker that extends BasePicker and uses the floating picker